### PR TITLE
System naming for docker-compose >= 2.0 fixed

### PR DESCRIPTION
**Pull request by pseusys**
As you might notice they started naming releases using not capitalized system name starting from `docker-compose` v2.0.0 ([ref](https://github.com/docker/compose/releases/tag/v2.0.0)).  
This commit adds version check and fixes system name (returned by `uname -s`).

### CODE REVIEW

**Comment by pseusys** [comment]
Solves #332 (yes, it's still an issue)

File: src/install.ts [hunk 1]
```diff
@@ -17,9 +17,10 @@ export async function runCommand(command: string): Promise<string> {
 }
 
 async function installOnLinux(version: string): Promise<string> {
-  const system = runCommand('uname -s')
-  const hardware = runCommand('uname -m')
-  const url = `https://github.com/docker/compose/releases/download/${version}/docker-compose-${await system}-${await hardware}`
+  const system = await runCommand('uname -s')
+  const hardware = await runCommand('uname -m')
+  const casedSystem = version.startsWith('2') ? `${system[0].toUpperCase()}${system.slice(1)}` : system
+  const url = `https://github.com/docker/compose/releases/download/${version}/docker-compose-${casedSystem}-${hardware}`
```
Comment (KengoTODA) [inline src/install.ts:23]: better to keep the `hardware` lazy:

```suggestion
  const hardware = runCommand('uname -m')
  const casedSystem = version.startsWith('2') ? `${system[0].toUpperCase()}${system.slice(1)}` : system
  const url = `https://github.com/docker/compose/releases/download/${version}/docker-compose-${casedSystem}-${await hardware}`
```

File: src/install.ts [hunk 2]
```diff
@@ -17,9 +17,10 @@ export async function runCommand(command: string): Promise<string> {
 }
 
 async function installOnLinux(version: string): Promise<string> {
-  const system = runCommand('uname -s')
-  const hardware = runCommand('uname -m')
-  const url = `https://github.com/docker/compose/releases/download/${version}/docker-compose-${await system}-${await hardware}`
+  const system = await runCommand('uname -s')
+  const hardware = await runCommand('uname -m')
+  const casedSystem = version.startsWith('2') ? `${system[0].toUpperCase()}${system.slice(1)}` : system
```
Comment (KengoTODA) [inline src/install.ts:22]: `version.startsWith('2')` will not work for v3 or later major versions.
Could you consider making it more strict, just like `parseInt(version.split('.')[0], 10) >= 2`?

**Review by KengoTODA — CHANGES_REQUESTED** [review]
Thanks for sending a PR! I left a few comments, please check them. :)

File: src/install.ts (hunk 2 above)
Comment (pseusys) [reply src/install.ts:22]: We still don't know release naming convention for v3 onward however.

File: src/install.ts (hunk 1 above)
Comment (pseusys) [reply src/install.ts:23]: Actually I don't see any difference, so be it

File: src/install.ts (hunk 2 above)
Comment (pseusys) [reply src/install.ts:22]: But you're right that it's highly likely that more recent format will stay.

File: src/install.ts [hunk 3]
```diff
@@ -17,9 +17,10 @@ export async function runCommand(command: string): Promise<string> {
 }
 
 async function installOnLinux(version: string): Promise<string> {
-  const system = runCommand('uname -s')
+  const system = await runCommand('uname -s')
   const hardware = runCommand('uname -m')
-  const url = `https://github.com/docker/compose/releases/download/${version}/docker-compose-${await system}-${await hardware}`
+  const casedSystem = parseInt(version.split('.')[0], 10) >= '2' ? `${system[0].toUpperCase()}${system.slice(1)}` : system
```
Comment (KengoTODA) [inline src/install.ts:22]: let's keep type-safety :)

```suggestion
  const casedSystem = parseInt(version.split('.')[0], 10) >= 2 ? `${system[0].toUpperCase()}${system.slice(1)}` : system
```

**Review by KengoTODA — CHANGES_REQUESTED** [review]

**Comment by pseusys** [comment]
Should I add new versions tests in `__tests__` dir, by the way?

File: src/install.ts (hunk 3 above)
Comment (pseusys) [reply src/install.ts]: Sure, my fault

**Comment by KengoTODA** [comment]
Yes, we should have tests for changes. Could you update the version for the integration test?  It can cover more than the unit test.
https://github.com/KengoTODA/actions-setup-docker-compose/blob/861c11a3c24fc1ec313191f5e5fed3be386f1069/.github/workflows/test.yml#L36

**Comment by KengoTODA** [comment]
From the v2 release, they added `v` as a prefix of their tag, so this action cannot download from GitHub Releases. This is why the integration test is failing.

https://github.com/KengoTODA/actions-setup-docker-compose/runs/6052564128?check_suite_focus=true#step:4:83

**Review by KengoTODA — COMMENTED** [review]
LGTM, please run `npm run all` and commit changes in the `dist` file. It will update optimized code and make the integration test success.

**Comment by pseusys** [comment]
I'm actually sorry for bothering you, but it appears to work correctly despite the platform names's first letter register. The only problem seems to be in version name (that starts from 'v' after docker-compose 2.0.0). I can make it append the letter automatically if it doesn't exist or I guess a piece of documentation in README should work quite as well.

File: src/install.ts [hunk 4]
```diff
@@ -19,6 +19,7 @@ export async function runCommand(command: string): Promise<string> {
 async function installOnLinux(version: string): Promise<string> {
   const system = runCommand('uname -s')
   const hardware = runCommand('uname -m')
+  version = parseInt(version.split('.')[0], 10) >= 2 ? `v${version}` : version
```
Comment (KengoTODA) [inline src/install.ts:22]: Sorry but I found some users [already use `v2.x.x` as version](https://github.com/xhaiker/rust-release.action/blob/d9f674cfe1789e23c5657acd808b1fafb1f8e0b1/.github/workflows/test.yaml#L20-L23), so better to support both `2.x.x` and `v2.x.x`. We can realize it by something like as follows:

```suggestion
  if (!version.startsWith('v') && parseInt(version.split('.')[0], 10) >= 2) {
    version = `v${version}`
  }
```

Please do not forget running `npm run all` before you make a commit :)

**Review by KengoTODA — CHANGES_REQUESTED** [review]

File: src/install.ts (hunk 4 above)
Comment (pseusys) [reply src/install.ts]: Yes, that what I meant saying about automatically appending 'v' if needed :)

**Review by KengoTODA — APPROVED** [review]
LGTM. Thanks for your contribution!

**Pull request merged**